oarepo_oidc_einfra/resources.py

```diff
@@ -0,0 +1,84 @@
+#
+# Copyright (C) 2024 CESNET z.s.p.o.
+#
+# oarepo-oidc-einfra  is free software; you can redistribute it and/or
+# modify it under the terms of the MIT License; see LICENSE file for more
+# details.
+#
+
+"""REST resources."""
+from datetime import datetime
+
+from flask import current_app, g, request
+from flask_resources import Resource, ResourceConfig, route
+from invenio_access import Permission, action_factory
+from invenio_files_rest.storage import PyFSFileStorage
+from invenio_records_resources.services.errors import PermissionDeniedError
+
+from oarepo_oidc_einfra.tasks import update_from_perun_dump
+
+upload_dump_action = action_factory("upload-oidc-einfra-dump")
+
+
+class OIDCEInfraResourceConfig(ResourceConfig):
+    """Configuration for the REST API."""
+
+    blueprint_name = "oarepo_oidc_einfra"
+    """Blueprint name."""
+
+    url_prefix = "/oidc-einfra"
+    """URL prefix for the resource, will be at /api/oidc-einfra."""
+
+    routes = {"upload-dump": "/dumps/upload"}
+    """Routes for the resource."""
+
+
+class OIDCEInfraResource(Resource):
+    """REST API for the EInfra OIDC."""
+
+    def __init__(self, config=None):
+        """Initialize the resource."""
+        super(OIDCEInfraResource, self).__init__(
+            config=config or OIDCEInfraResourceConfig()
+        )
+
+    def create_url_rules(self):
+        """Create URL rules for the resource."""
+        routes = self.config.routes
+        return [
+            route("POST", routes["upload-dump"], self.upload_dump),
+        ]
+
+    def upload_dump(self):
+        """Upload a dump of the EInfra data.
+
+        The dump will be uploaded to the configured location (EINFRA_DUMP_DATA_URL inside config)
+        and then processed by a celery synchronization task.
+
+        The caller must have the permission to upload the dump (upload-oidc-einfra-dump action
+        that can be assigned via invenio access commandline tool).
+        """
+        if not Permission(upload_dump_action).allows(g.identity):
```

Comment:
Consider stronger verification (cert-based? whitelist, signature checking) of the request